### DISCUSSION TITLE
Ranged creature AI won't attack target behind riot shield

### DISCUSSION THREAD

**Discussion by A2ur3** [FakeFishGames/Barotrauma#14871] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

https://github.com/user-attachments/assets/4f57848a-a53c-49de-a04e-30d2925d88be

### Reproduction steps

1. spawn creature Piratehuman01 from mod EuropaWar
2. hide behind a riot shield
3. creature wont attack even they can destroy the riot shield after a few shots

### Bug prevalence

Just once

### Single player or multiplayer?

Single player

### Version

v1.5.9.2 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by A2ur3** [comment]
https://github.com/user-attachments/assets/696f5eb7-3cb7-4e29-87a4-5e9fec88a77a

**Comment by itchyOwl** [comment]
Ticketed, thanks for reporting it.

**Comment by itchyOwl** [comment]
Couldn't repro the issue in the vanilla game. Tested in v1.5.9.2 (Summer Update Hotfix 2) and in the latest release 1.6.17.0. Don't have access to mods at the office, but I can test with the mod later.

**Comment by itchyOwl** [comment]
Found the issue and addressed it in #5984. At first didn't realize the issue was about creature AI, even though it's clearly stated in the title. You can create humanoid enemies with creature AI like that, but the creature AI isn't as "smart" as the human AI. So I'd suggest implementing the hostile humans as npcs, like in the vanilla game (e.g. pirates, bandits, and cultists). That way they also just use the human character and inherit all the changes there might come. If you implement them as separate creatures, like done here, you'll have to maintain them separately too. Moreover, they can't do stuff like healing, repairing, arresting, fixing etc. And the behavior overall is a bit different, like here the enemy just runs around constantly, like a madman (or a creature), whereas a human would stop to stand and shoot.

Anyway, should be fixed in the unstable in the near future.

**Comment by Rokvach** [comment]
Fixed and tested the fix to be working correctly. Fix coming in the next update. Closing discussion.
